p2p_wallet/Scenes/Main/Buy/BuyViewModel.swift

```diff
@@ -229,16 +246,67 @@ final class BuyViewModel: ObservableObject {
                 self.areMethodsLoading = false
             }
         }
+        
+        getBuyAvailability()
+    }
+    
+    private func getBuyAvailability() {
+        Task {
+            let ipInfo = try await moonpayProvider.ipAddresses()
```

Comment:
can we not use moonpay API directly, but through a Service.
protocol BuyService {
  fun country() -> BuyCountry
}

1. so we won't be dependent on Moonpay API change in every part of the code, but only in BuyService
2. we will remove Moonpay dependency from the file

Reply:
we don't have enough time for this, because of blocking AppStore release